### PR TITLE
Add `visible_hostname`, `via`, `httpd_suppress_version_string` and `forwarded_for` parameters

### PR DESCRIPTION
**Pull request by SourceDoctor**
set visible_hostname for Error Pages
also configure other visibility Options

### CODE REVIEW

File: templates/squid.conf.header.erb [hunk 1]
```diff
@@ -27,3 +27,12 @@ max_filedescriptors           <%= @max_filedescriptors %>
 workers           <%= @workers %>
 
 <% end -%>
+
+<% if @visible_hostname -%>
+visible_hostname              <%= @visible_hostname -%>
+<% end -%>
+via                           <%= (@via == 'on' || @via == true)?'on':'off' %>
```
Comment (alexjfisher) [inline templates/squid.conf.header.erb]: It can't be `on` as `via` is a Boolean.

File: manifests/params.pp [hunk 2]
```diff
@@ -7,6 +7,10 @@
   $ensure_service                = 'running'
   $enable_service                = true
   $cache_mem                     = '256 MB'
+  $visible_hostname              = 'mypc@workgroup'
+  $via                           = true
```
Comment (alexjfisher) [inline manifests/params.pp]: This *is* the default in squid, but perhaps it'd be better to default to `undef` and not write anything into the configuration file by default.

That way, nothing will break for users who are already setting these options by using `extra_config_sections`.

File: manifests/init.pp [hunk 3]
```diff
@@ -1,6 +1,8 @@
 class squid (
   String            $access_log                    = $squid::params::access_log,
   Pattern[/\d+ MB/] $cache_mem                     = $squid::params::cache_mem,
+  Boolean           $httpd_suppress_version_string = $squid::params::httpd_suppress_version_string,
```
Comment (alexjfisher) [inline manifests/init.pp]: I think these two could also be made `Optional[Boolean]`?

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -44,6 +48,10 @@
             config: '/tmp/squid.conf',
             cache_mem: '1024 MB',
             memory_cache_shared: 'on',
+            visible_hostname: 'testhost',
+            via: 'off',
+            httpd_suppress_version_string: 'on',
+            forwarded_for: 'off',
```
Comment (alexjfisher) [inline spec/classes/init_spec.rb]: String `off` is not a boolean.

**Comment by alexjfisher** [comment]
@SourceDoctor Could you look at the test failures?

**Comment by SourceDoctor** [comment]
@alexjfisher 
everything done, something else to handle or ready to merge?

File: spec/classes/init_spec.rb [hunk 5]
```diff
@@ -32,6 +32,7 @@
         it { is_expected.to contain_concat_fragment('squid_header').with_content(%r{^access_log\s+daemon:/var/log/#{squid_name}/access.log\s+squid$}) }
         it { is_expected.to contain_concat_fragment('squid_header').with_content(%r{^cache_mem\s+256 MB$}) }
         it { is_expected.to contain_concat_fragment('squid_header').with_content(%r{^maximum_object_size_in_memory\s+512 KB$}) }
+
```
Comment (traylenator) [inline spec/classes/init_spec.rb]: Unescesary blank line.

**Comment by traylenator** [comment]
@SourceDoctor thanks for all the patches. Time a release after that lot.

**Pull request merged**